handshake.go

```diff
@@ -157,6 +158,9 @@ func (h *ClientHandshake) HandleNewSessionTicket(hm *HandshakeMessage) (PreShare
 		IsResumption: true,
 		Identity:     tkt.Ticket,
 		Key:          h.Context.resumptionSecret,
+		ReceivedAt:   time.Now(),
+		ExpiresAt:    time.Now().Add(time.Duration(tkt.TicketLifetime) * time.Second),
```

Comment:
You should check ExpiresAt when you pull the ticket and if it's expired, don't send it.  It would also be good to go ahead and delete it, but you'll need to add a `Delete(string)` method to `ClientSessionCache` for that.